### DISCUSSION TITLE
Does auto-instrumentation create spans during transform operation on KStream

### DISCUSSION THREAD

**Discussion by itsmaneesh** [open-telemetry/opentelemetry-java-instrumentation#5704] [Q&A]
Hi,
I have a requirement where I need to create spans while performing transform operation on KStream object. 
Auto-instrumentation doesn't seem to create spans automatically during this transform operation. 
for e.g.:
streamsBuilder.stream(myTopic.getTopicName()).transform(
                // needed a span for the below TransformSupplier 
                () -> new Transformer<Object, Object, KeyValue<Object, Object>>() {
                    private ProcessorContext context;

                    @Override
                    public void init(final ProcessorContext context) {
                        this.context = context;
                    }

                    @Override
                    public KeyValue<Object, Object> transform(Object key, Object value) {
                        // Logic to update the key and value
                        return new KeyValue<>(key, value);
                    }

                    @Override
                    public void close() {
                    }
                }
        );
I have observed that there no span created automatically with auto-instrumentation. But, wanted to check your advice to achieve the requirement.

**Comment by trask** [comment]
hi @itsmaneesh! you can capture additional spans using `@WithSpan` or using the OpenTelemetry API

**Reply by itsmaneesh** [reply]
@trask Thank you for the response. Yes, thought so about @WithSpan .
